### PR TITLE
First bunch of localization commits

### PR DESCRIPTION
**Pull request by fhebel**

### CODE REVIEW

**Comment by hishamco** [comment]
It's glad to see you PR @fhebel, I was waiting for yesterday in Gitter after I read your email

**Comment by hishamco** [comment]
Before I review the code, could you please mention what you are trying to fix or enhance, if there more than one thing small PRs will be better

**Comment by thiennn** [comment]
The code is pretty good. The change can be separate into to PRs: one for contact areas translation and one for the remaining. But it is ok, I can understand all the changes.

**Comment by hishamco** [comment]
@thiennn let me review it then ...

File: src/Modules/SimplCommerce.Module.Localization/EFStringLocalizer.cs [hunk 1]
```diff
@@ -61,17 +61,17 @@ private string GetString(string name)
...
+                AutoRegisterNewString(name, culture);
+            }
 
             return value;
         }
 
         private void AutoRegisterNewString(string name, string culture)
         {
-            if (culture == "en-US") //GlobalConfiguration.DefaultCulture
+            if (culture != "en-US") //GlobalConfiguration.DefaultCulture
```
Comment (hishamco) [inline src/Modules/SimplCommerce.Module.Localization/EFStringLocalizer.cs:74]: Could please let me know why you negate the condition?

Comment (fhebel) [reply src/Modules/SimplCommerce.Module.Localization/EFStringLocalizer.cs:74]: I need for all languages, which are not en-US the possibility to choose it on the translations page and to enter the respective translations.
In my personal case I'm working with de-DE and en-GB, so I can enter the translations for those two languages ...

Comment (fhebel) [reply src/Modules/SimplCommerce.Module.Localization/EFStringLocalizer.cs:74]: My initial question regarding this change was:
Can everybody agree that the standard texts in the application can be considered en-US (e.g. @localizer[“Cancel”]), which means that all other languages can be considered a translation?

Comment (hishamco) [reply src/Modules/SimplCommerce.Module.Localization/EFStringLocalizer.cs:74]: > Can everybody agree that the standard texts in the application can be considered en-US (e.g. @localizer[“Cancel”]), which means that all other languages can be considered a translation?

Agree, that's the default

File: src/Modules/SimplCommerce.Module.Contacts/Areas/Contacts/Controllers/ContactAreaTranslationApiController.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+            var entityType = contactArea.GetType().Name;
+
+            var localizeProperties = _localizedContentPropertyRepository.Query().Where(x => x.EntityId == contactArea.Id
+                && x.EntityType == entityType && x.CultureId == culture);
+
+            var model = new ContactAreaTranslationForm
+            {
+                DefaultCultureName = contactArea.Name,
+                Name = localizeProperties.FirstOrDefault(x => x.ProperyName == nameof(contactArea.Name))?.Value,
```
[the remark this answers is not in the document]
Comment (fhebel) [reply src/Modules/SimplCommerce.Module.Contacts/Areas/Contacts/Controllers/ContactAreaTranslationApiController.cs:44]: Good point, I have to check where I copied the controller from. There will be the same problem ...

**Comment by fhebel** [comment]
@thiennn @hishamco I believe I resolved all issues ...

**Comment by thiennn** [comment]
Thank you so much

**Comment by fhebel** [comment]
You're welcome. As soon as I my shop is in production I'll share more ...

**Pull request merged**